test/block-test.js

```diff
@@ -4,19 +4,72 @@
 
 'use strict';
 
-const {BloomFilter} = require('bfilter');
 const assert = require('bsert');
-const common = require('./util/common');
 const Block = require('../lib/primitives/block');
-const MerkleBlock = require('../lib/primitives/merkleblock');
-const consensus = require('../lib/protocol/consensus');
-const Network = require('../lib/protocol/network');
-const Script = require('../lib/script/script');
-const bip152 = require('../lib/net/bip152');
-const CompactBlock = bip152.CompactBlock;
-const TXRequest = bip152.TXRequest;
-const TXResponse = bip152.TXResponse;
+const MemBlock = require('../lib/primitives/memblock');
 
 describe('Block', function() {
   this.timeout(10000);
+
+  describe('Serialization', function() {
+    let block = null;
+    let raw = null;
+
+    it('should create a block from JSON', async () => {
+      block = Block.fromJSON({
+        version: 1,
+        prevBlock:
+          '0000000000000000000000000000000000000000000000000000000000000000',
+        merkleRoot:
+          '0101010101010101010101010101010101010101010101010101010101010101',
+        witnessRoot:
+          '0202020202020202020202020202020202020202020202020202020202020202',
+        treeRoot:
+          '0303030303030303030303030303030303030303030303030303030303030303',
+        reservedRoot:
+          '0404040404040404040404040404040404040404040404040404040404040404',
+        extraNonce:
+          '050505050505050505050505050505050505050505050505',
+        mask:
+          '0606060606060606060606060606060606060606060606060606060606060606',
+        time: 0,
+        bits: 0,
+        nonce: 0,
+        txs: [
+          {
+            version: 1,
+            locktime: 0,
+            inputs: [
+              {
+                sequence: 0,
+                prevout: {
+                  hash: Buffer.alloc(32, 7).toString('hex'),
+                  index: 0
+                },
+                witness: []
+              }
+            ],
+            outputs: [
+              {
+                value: 1234567,
+                address: 'rs1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqn6kda'
+              }
+            ]
+          }
+        ]
+      });
+      assert(block);
```

Comment:
How do you feel about asserting the individual attributes on the block primitive?

Reply:
Good idea, done at 3419d7cfece14db95cc899f996a351c8527f8387